packages/ui-utils/src/types.ts

```diff
@@ -117,6 +117,15 @@ export type ToolInvocation =
   | CoreToolCall<string, any>
   | CoreToolResult<string, any, any>;
 
+/**
+ * An attachment that can be sent along with a message.
+ */
+export interface Attachment {
+  name?: string;
+  mimeType?: string;
+  url: string;
```

Comment:
Would be good to add JSDoc for the fields (it's shown in the IDE automatically when you use those fields)

Reply:
Then we can also mention that "data" urls are supported and you do not necessarily have to have an uploaded URL

Reply:
good callout, done!